File: app/src/main/java/ru/nikshlykov/englishwordsapp/ui/fragments/DeleteSubgroupDialogFragment.java
package ru.nikshlykov.englishwordsapp.ui.fragments;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import ru.nikshlykov.englishwordsapp.R;

public class DeleteSubgroupDialogFragment extends DialogFragment {

    // Тег для логирования.
    private static final String LOG_TAG = "DeleteSubgroupDF";

    // Сообщение о том, что удаление подтверждено.
    public static final String DELETE_MESSAGE = "Delete";

    // Интерфейс для взаимодействия с Activity.
    private DeleteSubgroupListener deleteSubgroupListener;

    public interface DeleteSubgroupListener {
        void deleteMessage(String message);
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        // Присваиваем слушатель.
        // TODO разобраться с удалением подгруппы. Т.к. этот слушатель больше не работает.
        // deleteSubgroupListener = (DeleteSubgroupListener) context;

        // TODO просмотреть ещё диалоговые фрагменты, в которых могут быть слушатели.
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new AlertDialog.Builder(getActivity())
                .setTitle(R.string.dialog___delete_subgroup___title)
                .setMessage(R.string.dialog___delete_subgroup___message)
                .setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        // Отправляем Activity сообщение о том, что удаление подтверждено.
                        deleteSubgroupListener.deleteMessage(DELETE_MESSAGE);
                    }
                })
                .setNegativeButton(R.string.no, null)
                .create();
    }
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/ui/fragments/DeleteSubgroupDialogFragment.kt
package ru.nikshlykov.englishwordsapp.ui.fragments

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import ru.nikshlykov.englishwordsapp.R

class DeleteSubgroupDialogFragment : DialogFragment() {

  private var deleteSubgroupListener: DeleteSubgroupListener? = null

  interface DeleteSubgroupListener {
    fun deleteMessage(message: String?)
  }

  // TODO просмотреть ещё диалоговые фрагменты, в которых могут быть слушатели.
  //  Т.к. раньше были acitivity, которые их обрабатывали. А сейчас всё перешло на фрагменты.
  fun setDeleteSubgroupListener(deleteSubgroupListener: DeleteSubgroupListener) {
    this.deleteSubgroupListener = deleteSubgroupListener
  }

  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    return AlertDialog.Builder(requireActivity())
      .setTitle(R.string.dialog___delete_subgroup___title)
      .setMessage(R.string.dialog___delete_subgroup___message)
      .setPositiveButton(R.string.yes) { dialog, which -> // Отправляем Activity сообщение о том, что удаление подтверждено.
        deleteSubgroupListener!!.deleteMessage(DELETE_MESSAGE)
      }
      .setNegativeButton(R.string.no, null)
      .create()
  }

  companion object {
    // Тег для логирования.
    private const val LOG_TAG = "DeleteSubgroupDF"

    // Сообщение о том, что удаление подтверждено.
    const val DELETE_MESSAGE = "Delete"
  }
}